write-fonts/src/graph/splitting.rs

```diff
@@ -0,0 +1,402 @@
+//! splitting layout subtables
+
+use std::collections::{HashMap, HashSet};
+
+use font_types::{GlyphId, Offset16};
+
+use super::{Graph, ObjectId};
+use crate::tables::layout as wlayout;
+use crate::types::FixedSize;
+use crate::write::TableData;
+
+use crate::read::tables::gpos as rgpos;
+use crate::read::tables::layout as rlayout;
+
+const MAX_TABLE_SIZE: usize = u16::MAX as usize;
+
+pub(super) fn split_pair_pos(graph: &mut Graph, lookup: ObjectId) {
+    split_subtables(graph, lookup, split_pair_pos_subtable)
+}
+
+/// A common impl handling updating the lookuup with the new subtables
+fn split_subtables(
+    graph: &mut Graph,
+    lookup: ObjectId,
+    split_fn: fn(&mut Graph, ObjectId) -> Option<Vec<ObjectId>>,
+) {
+    let data = graph.objects.remove(&lookup).unwrap();
+    debug_assert!(data.reparse::<rgpos::PositionLookup>().is_ok());
```

Comment:
This function is called `split_subtables` and doesn’t seem to actually depend on it being a GPOS subtable. This feels like a bit of a footgun if/when we extend to GSUB.

Reply:
My understanding is that table splitting is only relevant for GPOS, where individual subtables can exceed 64k bytes. in GSUB the main problem is with having so many subtables that you get offset overflows, and that's handled by extension promotion.

That said I agree I should make that more explicit :)